src/field_theory/adjoin.lean

```diff
@@ -250,10 +252,39 @@ adjoin_adjoin_left _ _ _
 lemma adjoin_simple_comm (β : E) : ↑F⟮α⟯⟮β⟯ = (↑F⟮β⟯⟮α⟯ : intermediate_field F E) :=
 adjoin_adjoin_comm _ _ _
 
+lemma adjoin_simple_to_subalgebra_of_integral (hα : is_integral F α) :
+  (F⟮α⟯).to_subalgebra = algebra.adjoin F {α} :=
+begin
+  apply adjoin_eq_algebra_adjoin,
+  intros x hx,
+  by_cases x = 0,
+  { rw h,
+    rw inv_zero,
+    exact subalgebra.zero_mem (algebra.adjoin F {α}) },
+  let ϕ := alg_equiv.adjoin_singleton_equiv_adjoin_root_minimal_polynomial F α hα,
+  let inv := (@adjoin_root.field F _ _ (minimal_polynomial.irreducible hα)).inv,
+  suffices key : ↑(ϕ.symm (inv (ϕ (⟨x, hx⟩)))) = x⁻¹,
+  { rw ←key,
+    exact subtype.mem (ϕ.symm (inv (ϕ ⟨x, hx⟩))) },
+  suffices : ϕ ⟨x, hx⟩ * inv (ϕ ⟨x, hx⟩) = 1,
+  { apply eq_inv_of_mul_right_eq_one,
+    apply_fun ϕ.symm at this,
+    rw [alg_equiv.map_one, alg_equiv.map_mul, alg_equiv.symm_apply_apply] at this,
+    rw [←subsemiring.coe_one, ←this, subsemiring.coe_mul, subtype.coe_mk] },
+  rw field.mul_inv_cancel,
+  intro key,
+  rw ← alg_equiv.map_zero ϕ at key,
+  replace key := ϕ.injective key,
+  apply h,
+  change ↑(⟨x, hx⟩ : algebra.adjoin F {α}) = _,
+  rw key,
+  refl,
+end
```

Comment:
```suggestion
-- TODO: develop the API for `subalgebra.is_field_of_algebraic` so it can be used here
lemma adjoin_simple_to_subalgebra_of_integral (hα : is_integral F α) :
  (F⟮α⟯).to_subalgebra = algebra.adjoin F {α} :=
begin
  apply adjoin_eq_algebra_adjoin,
  intros x hx,
  by_cases x = 0,
  { rw [h, inv_zero], exact subalgebra.zero_mem (algebra.adjoin F {α}) },

  let ϕ := alg_equiv.adjoin_singleton_equiv_adjoin_root_minimal_polynomial F α hα,
  let inv := (@adjoin_root.field F _ _ (minimal_polynomial.irreducible hα)).inv,
  suffices : ϕ ⟨x, hx⟩ * inv (ϕ ⟨x, hx⟩) = 1,
  { convert subtype.mem (ϕ.symm (inv (ϕ ⟨x, hx⟩))),
    refine (eq_inv_of_mul_right_eq_one _).symm,
    apply_fun ϕ.symm at this,
    rw [alg_equiv.map_one, alg_equiv.map_mul, alg_equiv.symm_apply_apply] at this,
    rw [←subsemiring.coe_one, ←this, subsemiring.coe_mul, subtype.coe_mk] },

  rw field.mul_inv_cancel (mt (λ key, _) h),
  rw ← ϕ.map_zero at key,
  change ↑(⟨x, hx⟩ : algebra.adjoin F {α}) = _,
  rw [ϕ.injective key, submodule.coe_zero]
end
```